### ISSUE TITLE
Option group support

### ISSUE THREAD

**Issue by smilingpeanut** [optimistdigital/nova-multiselect-field#4]
Would it be possible to have this support `<optgroups>`, which would bring parity with the native select field?

Thank you!

**Comment by Tarpsvo** [comment]
Certainly possible, but requires some thought and development. We'll look into it in the future. PR's are always welcome though.

**Comment by pzmarzly** [comment]
Vue multiselect, which is used internally, [has support for it](https://vue-multiselect.js.org/#sub-option-groups). Meanwhile Nova's native Select [also supports it, and has an established API convention](https://stackoverflow.com/a/59554040/5108318). Laravel Collections have a `groupBy` method, which can be used to transform the data format like so:

```php
$c = collect([
  ['name' => 'a', 'group' => 'x'],
  ['name' => 'b', 'group' => 'y'],
  ['name' => 'c', 'group' => 'y'],
]);

$c->groupBy('group')
```

```text
=> Illuminate\Support\Collection {#1257
     all: [
       "x" => Illuminate\Support\Collection {#1255
         all: [
           [
             "name" => "a",
             "group" => "x",
           ],
         ],
       },
       "y" => Illuminate\Support\Collection {#1256
         all: [
           [
             "name" => "b",
             "group" => "y",
           ],
           [
             "name" => "c",
             "group" => "y",
           ],
         ],
       },
     ],
   }
```

**Comment by Tarpsvo** [comment]
Done today and released in version 1.8.0. Thanks @pzmarzly for bringing the data together for me!  Had I know it's that easy, I would've done it sooner.
